### PR TITLE
docs: Add sequence diagram for xDS error_detail NACK example

### PR DESCRIPTION
**Pull request by eric846**
Signed-off-by: eric846 <56563761+eric846@users.noreply.github.com>

Commit Message: docs: Add sequence diagram for xDS error_detail NACK example
Additional Description: n/a
Risk Level: Low: doc formatting
Testing: n/a
Docs Changes: converted text to a diagram
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: api/diagrams/error-detail-nack.svg [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8" standalone="no"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN" "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd"><svg xmlns="http://www.w3.org/2000/svg" width="1092" height="678" xmlns:xlink="http://www.w3.org/1999/xlink"><source><![CDATA[participant Envoy as C
+participant Management Server as S
+
+C->S: {resource_names={A}}
+S->C: {resources={A}, version=1, nonce=1}
+C->S: {resource_names={A}, version=1, nonce=1 (this is an ACK)}
+C->S: {resource_names={A, B}, version=1, nonce=1}
+Note over C: adding subscription\nto B
```
Comment (htuch) [inline api/diagrams/error-detail-nack.svg:8]: I think in all the "Note over"s that they are below the line they are referring to. E.g. here, the addition of subscription to B by the client happens in the line above when `{A, B}` is sent.

**Review by htuch — CHANGES_REQUESTED** [review]
Thanks @eric846, this should help clarify, just one nit.
/wait

File: api/diagrams/error-detail-nack.svg (hunk 1 above)
Comment (eric846) [reply api/diagrams/error-detail-nack.svg:7]: Done, should be clearer now.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**